sklearn/manifold/spectral_embedding_.py

```diff
@@ -288,11 +288,12 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         laplacian = check_array(laplacian, dtype=np.float64,
                                 accept_sparse=True)
         laplacian = _set_diag(laplacian, 1, norm_laplacian)
+        laplacian = laplacian + 1e-5 * sparse.eye(laplacian.shape[0])
         ml = smoothed_aggregation_solver(check_array(laplacian, 'csr'))
         M = ml.aspreconditioner()
         X = random_state.rand(laplacian.shape[0], n_components + 1)
         X[:, 0] = dd.ravel()
-        lambdas, diffusion_map = lobpcg(laplacian, X, M=M, tol=1.e-12,
+        lambdas, diffusion_map = lobpcg(laplacian, X, M=M, tol=1.e-5,
```

Comment:
Is this change necessary for tests to pass?

Reply:
tol=1.e-12 is unreasonably small here, unnecessary for the quality of the embedding, and making lobpcg run longer for no reason and possibly even crash, until https://github.com/scipy/scipy/pull/10621 is released